corehq/apps/hqwebapp/views.py

```diff
@@ -1348,3 +1351,62 @@ def log_email_event(request, secret):
     handle_email_sns_event(message)
 
     return HttpResponse()
+
+
+@method_decorator(require_superuser, name="dispatch")
+class OauthApplicationRegistration(BasePageView):
+    urlname = 'oauth_application_registration'
+    page_title = "Oauth Application Registration"
+    template_name = "hqwebapp/oauth_application_registration_form.html"
+
+    @property
+    def page_url(self):
+        return reverse(self.urlname)
+
+    @property
+    def base_application_form(self):
+        return modelform_factory(
+            get_application_model(),
```

Comment:
what's the benefit of generating the form dynamically versus explicitly creating a form class? I guess less cruft at the expense of some developer-familiarity?

Reply:
Or maybe this is necessary because the application model is swappable?

Reply:
I actually wanted to directly reuse the form that `django-oauth-toolkit` provided, but they generate it like this. Happy to define these two as actual forms, although I'm not sure what the benefit would be since there's no customization on either of them.